src/api/entitlements.js

```diff
@@ -133,21 +161,46 @@ export class Entitlements {
   /**
    * Returns the first matching entitlement for the specified product,
    * optionally also matching the specified source.
+   *
+   * Returns non-metering entitlements if possible, to avoid using
+   * metered reads unnecessarily.
+   *
    * @param {?string} product
    * @param {string=} source
    * @return {?Entitlement}
    */
   getEntitlementFor(product, source) {
-    if (product && this.entitlements.length > 0) {
-      for (let i = 0; i < this.entitlements.length; i++) {
-        if (
-          this.entitlements[i].enables(product) &&
-          (!source || source == this.entitlements[i].source)
-        ) {
-          return this.entitlements[i];
-        }
+    if (!product) {
+      return null;
+    }
+
+    // Sort non-metering entitlements first, so they are returned if possible,
+    // to avoid using metered reads unnecessarily.
+    const entitlements = this.entitlements.slice().sort((a, b) => {
+      if (a.source === b.source) {
+        return 0;
+      }
+
+      if (a.source === GOOGLE_METERING_SOURCE) {
+        return 1;
+      }
+
+      if (b.source === GOOGLE_METERING_SOURCE) {
+        return -1;
+      }
+
+      return 0;
+    });
+
+    for (let i = 0; i < entitlements.length; i++) {
+      if (
+        entitlements[i].enables(product) &&
+        (!source || source == entitlements[i].source)
```

Comment:
add comment to explain what you're checking for here

Reply:
I refactored this code to (hopefully) be more descriptive